### DISCUSSION TITLE
Registering logger/trainer inside nested PL models

### DISCUSSION THREAD

**Discussion by CompRhys** [Lightning-AI/pytorch-lightning#13144] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
`You are trying to 'self.log()' but the 'self.trainer' reference is not registered on the model yet. This is most likely because the model hasn't been passed to the 'Trainer'`

I have a PL module that I am now incorporating inside another PL module as part of a more complicated setup. Inside the sub module there are still various things that I would like to log but as it's not being trained directly instead via a trainer on the overall module the self.trainer is not set. Is it okay to just pass self.trainer of the overall model as an additional kwarg to the sub model or could this have unforeseen consequences?

**Comment by akihironitta** [comment]
Hi @CompRhys! #13638 might be relevant to your use case. It enables `LightningModule`s of a parent `LightningModule` to have reference to `Trainer`, so with this change, you shouldn't see the error you report anymore:
> `You are trying to 'self.log()' but the 'self.trainer' reference is not registered on the model yet. This is most likely because the model hasn't been passed to the 'Trainer'`

Please mind that the change is currently available only in master but will be available in the next minor release v1.7.
